path.py

```diff
@@ -363,6 +363,18 @@ def drive(self):
         .. seealso:: :meth:`basename`, :func:`os.path.basename`
         """)
 
+    def setext(self, new_ext):
+        """ Set a new extension for this file.
+        If the new extension does not start with a dot,
```

Comment:
Zen of Python says there should be (preferably) one right way to do things. I suggest rejecting inputs with dots or without. Since `os.splitext` gives the extension with a dot, that would be the preferred syntax. It would also be acceptable not to check, but to assume, and require the user provide such a value.